src/Controls/src/Core/NavigationPage/NavigationPage.cs

```diff
@@ -869,11 +869,12 @@ protected override Task OnPushAsync(Page root, bool animated)
 				return Owner.SendHandlerUpdateAsync(animated,
 					() =>
 					{
+						// Move the SendNavigating here so that it's fired prior to the stack being modified
```

Comment:
Can include in the comment something like:
`"This ensures consistent event ordering across all platforms (iOS, Catalyst, Android, Windows)"`

Reply:
@jsuarezruiz I’ve added the comment as suggested.